src/Engine/Manager/SoundManager.cpp

```diff
@@ -75,38 +70,86 @@ void SoundManager::CheckError(PaError err) {
 void SoundManager::Update(float deltaTime) {
 
     // Number of samples to process dependant on deltaTime
-    int numSamples = int(SAMPLE_RATE * deltaTime);
+    /*unsigned int frameSamples = int(SAMPLE_RATE * deltaTime);
+    if (frameSamples > CHUNK_SIZE) {
+        //Log() << "SoundManager::Update: Frame drop!\n";
+        frameSamples = CHUNK_SIZE;
+    }
+    targetSample += frameSamples;
+
+    // Update sound.
+    while (currentSample < targetSample) {
+        // Process Samples.
+        if (processedSamples == 0) {
+            ProcessSamples();
+            processedSamples = CHUNK_SIZE;
+        }
+
+        // Play samples.
+        unsigned int sampleCount = std::min(targetSample - currentSample, processedSamples);
+        unsigned int index = (CHUNK_SIZE - processedSamples) * 2;
+        Pa_WriteStream(stream, &processedBuffer[index], sampleCount);
+        processedSamples -= sampleCount;
+        currentSample += sampleCount;
+    }*/
+}
 
+void SoundManager::ProcessSamples() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'ProcessSamples' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)